providers/reddit/reddit_test.go

```diff
@@ -0,0 +1,28 @@
+package reddit
+
+import (
+	"fmt"
+	"testing"
+)
+
+// TestBuildURI makes sure the BuildURI method builds valid URIs.
+func TestBuildURI(t *testing.T) {
+	p := Provider{}
+	cases := []struct {
+		query, want string
+	}{
+		// Regular reddit search.
+		{"best startups",
+			fmt.Sprintf("https://www.reddit.com/search?q=%s", "best+startups")},
+		// Subreddit search.
+		{"/r/cscareerquestions best startups",
```

Comment:
If query is same as above, the query would be "https://www.reddit.com/r/cscareerquestions/search?q=&restrict_sr=on&sort=relevance&t=all"
which is still a valid query. Or did refer to something else?